### PR TITLE
CLI for percentiles-to-probabilities plugin

### PR DESCRIPTION
**Pull request by cgsandford**
IMPRO-296

CLI and bats tests for new plugin (see https://github.com/metoppv/improver/https://github.com/metoppv/improver/pull/402)

### CODE REVIEW

File: bin/improver-percentiles-to-probabilities [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to collapse cube coordinates and calculate percentiled data."""
+
+import argparse
+
+import iris
+import warnings
```
Comment (wxtim) [inline bin/improver-percentiles-to-probabilities]: This import isn't used

File: bin/improver-percentiles-to-probabilities [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+from improver.utilities.load import load_cube
+from improver.utilities.statistical_operations import \
+    ProbabilitiesFromPercentiles2D
+
+
+def main():
+    """ Load arguments and run ProbabilitiesFromPercentiles plugin """
+    parser = argparse.ArgumentParser(
+        description="Calculate probability from a percentiled field at a "
```
Comment (wxtim) [inline bin/improver-percentiles-to-probabilities]: I find the ascii art in the plugin really helpful when trying to understand this - could we add it here?

File: bin/improver-percentiles-to-probabilities (hunk 1 above)
Comment (cgsandford) [reply bin/improver-percentiles-to-probabilities]: You're right, it's not.  I wonder why the tests didn't pick this up?

**Comment by wxtim** [comment]
I've added a PR for my proposed changes for you at https://github.com/cgsandford/improver/pulls

**Comment by wxtim** [comment]
I have created a notebook at https://github.com/MetOffice/improver_aux/pull/5 to check these input and output data

**Comment by cgsandford** [comment]
Test data for basic.bats are stored in /scratch/tpilling/IMPRO-296/percentiles-to-probabilities/basic/ (copy /scratch/csand/IMPRO-296/basic_bats/).

File: tests/improver-percentiles-to-probabilities/02-basic.bats [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      "$IMPROVER_ACC_TEST_DIR/percentiles-to-probabilities/basic/enukx_orography.nc" \
+      "$TEST_DIR/output.nc"
+  [[ "$status" -eq 0 ]]
+
+  # Run nccmp to compare the output and kgo.
+  improver_compare_output "$TEST_DIR/output.nc" \
+      "$IMPROVER_ACC_TEST_DIR/percentiles-to-probabilities/basic/kgo.nc"
+  rm "$TEST_DIR/output.nc"
+  rmdir "$TEST_DIR"
+}
```
Comment (markysparks) [inline tests/improver-percentiles-to-probabilities/02-basic.bats]: Should we have another test to check that the '--new_name' argument is working? i.e.

# Run percentiles to probability processing and check it passes.
  run improver percentiles-to-probabilities "$IMPROVER_ACC_TEST_DIR/perc-prob/input.nc" \
  "$IMPROVER_ACC_TEST_DIR/perc-prob/sfc_altitude.nc" "$TEST_DIR/output.nc" \
  --new_name="Snow Prob"
  [[ "$status" -eq 0 ]]

  # Run nccmp to compare the output and kgo.
  improver_compare_output "$TEST_DIR/output.nc" \
      "$IMPROVER_ACC_TEST_DIR/perc-prob/change-name/kgo.nc"
  rm "$TEST_DIR/output.nc"
  rmdir "$TEST_DIR"

The above assumes you a have a folder in your test data directory called 'change-name' .

**Comment by cgsandford** [comment]
This is broken for the moment because the unit tests for IMPRO-296 plugin weren't merged in https://github.com/metoppv/improver/pull/402.  Pursuing a fix for that.  Bats tests are still good for review.

**Comment by wxtim** [comment]
This would produce more realistic KGO if the snow_level.nc had more than 3 percentiles, but it's not strictly necessary atm.

**Pull request merged**